app/src/main/java/com/github/orkest/View/EditProfileActivity.kt

```diff
@@ -0,0 +1,221 @@
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import coil.compose.rememberImagePainter
+import com.github.orkest.View.ui.theme.OrkestTheme
+import com.github.orkest.R
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.graphics.Color
+
+class EditProfileActivity : ComponentActivity() {
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContent {
+            editProfileSetting {
+                EditProfileScreen()
+            }
+        }
+    }
+}
+
+/**
+ * Function generating the screen
+ */
+@Composable
+fun editProfileSetting(content: @Composable () -> Unit) {
+    OrkestTheme {
+        // A surface container using the 'background' color from the theme
+        Surface(
+            modifier = Modifier.fillMaxSize(),
+            color = MaterialTheme.colorScheme.background
+        ) {
+            content()
+        }
+    }
+}
+
+/**
+ * Principal function in which we build the general structure of the activity
+ */
+@Composable
+fun EditProfileScreen() {
+    Column(modifier = Modifier.fillMaxHeight()){
+        topBar()
+        Divider()
+        mainBody()
+    }
+}
+
+/**
+ * Top part of the screen, containing:
+ * - "cancel" button on the top left to stop editing changes and returning to profile screen
+ * - "save" button saving changes to database, updating profile screen and returning to it
+ * - current profile picture
+ * - "edit picture" button to modify picture
+ */
+@Composable
+fun topBar() {
+    Column(
+        modifier = Modifier
+            .verticalScroll(rememberScrollState())
+            .padding(8.dp)
+    ) {
+        Row(
+            modifier = Modifier
+                .fillMaxWidth()
+                .padding(8.dp),
+            // this spaces out the "cancel" and "save" buttons
+            horizontalArrangement = Arrangement.SpaceBetween
+        ) {
+            // "cancel" clickable text (button)
+            Text(
+                text = "Cancel",
+                modifier = Modifier.clickable { /* TODO */ }
+            )
+
+            // "save" clickable text (button)
+            Text(
+                text = "Save",
+                modifier = Modifier.clickable { /* TODO */ }
+            )
+        }
+        // profile pic and edit button
+        EditProfileImage()
+    }
+}
+
+/**
+ * The larger part of the screen containing the fields to modify textual information
+ */
+@Composable
+fun mainBody() {
+    EditNameSection(name = "Name", default = "default name")
```

Comment:
I think we decided recently to only keep a username for users, no additional names, in order to not complicate the database architecture and research.

Reply:
say less